cfme/tests/infrastructure/test_instance_analysis.py

```diff
@@ -9,51 +9,100 @@
 from cfme.common.provider import cleanup_vm
 from cfme.configure import configuration, tasks
 from cfme.infrastructure import host
-from cfme.infrastructure.pxe import get_template_from_config
 from cfme.provisioning import do_vm_provisioning
 from cfme.web_ui import InfoBlock, Table, SplitTable, paginator, tabstrip as tabs, toolbar as tb
 from fixtures.pytest_store import store
-from utils import testgen, ssh
+from utils import testgen, ssh, safe_string, version
 from utils.conf import cfme_data
+from utils.log import logger
 from utils.wait import wait_for
-from utils.version import pick, LOWEST
+
+WINDOWS = {'id': "Red Hat Enterprise Windows", 'icon': 'linux_windows'}
+
+RPM_BASED = {
+    'rhel': {
+        'id': "Red Hat", 'release-file': '/etc/redhat-release', 'icon': 'linux_redhat',
+        'package': "kernel", 'install-command': "",  # We don't install stuff on RHEL
+        'package-number': 'rpm -qa | wc -l'},
+    'centos': {
+        'id': "CentOS", 'release-file': '/etc/centos-release', 'icon': 'linux_centos',
+        'package': 'iso-codes', 'install-command': 'yum install -y {}',
+        'package-number': 'rpm -qa | wc -l'},
+    'fedora': {
+        'id': 'Fedora', 'release-file': '/etc/fedora-release', 'icon': 'linux_fedora',
+        'package': 'iso-codes', 'install-command': 'dnf install -y {}',
+        'package-number': 'rpm -qa | wc -l'},
+    'suse': {
+        'id': 'Suse', 'release-file': '/etc/SuSE-release', 'icon': 'linux_suse',
+        'package': 'iso-codes', 'install-command': 'zypper install -y {}',
+        'package-number': 'rpm -qa | wc -l'},
+}
+
+DEB_BASED = {
+    'ubuntu': {
+        'id': 'Ubuntu 14.04', 'release-file': '/etc/issue.net', 'icon': 'linux_ubuntu',
+        'package': 'iso-codes',
+        'install-command': 'env DEBIAN_FRONTEND=noninteractive apt-get -y install {}',
+        'package-number': "dpkg --get-selections | wc -l"},
+    'debian': {
+        'id': 'Debian ', 'release-file': '/etc/issue.net', 'icon': 'linux_debian',
+        'package': 'iso-codes',
+        'install-command': 'env DEBIAN_FRONTEND=noninteractive apt-get -y install {}',
+        'package-number': 'dpkg --get-selections | wc -l'},
+}
 
 
 def pytest_generate_tests(metafunc):
     # Filter out providers without templates defined
-    argnames, argvalues, idlist = testgen.infra_providers(metafunc, 'provisioning')
+    argnames, argvalues, idlist = testgen.infra_providers(
+        metafunc, "vm_analysis_new", require_fields=True)
```

Comment:
Just fyi: https://github.com/RedHatQE/cfme_tests/blob/9d16d76e4ed86b214232aec7db2b2b9a911853db/utils/testgen.py#L331

`require_fields=` is True by default but that is not an issue, gonna merge it anyway.